packages/lib-classifier/src/components/Classifier/components/SubjectViewer/components/LightCurveViewer/LightCurveViewerContainer.js

```diff
@@ -0,0 +1,127 @@
+import asyncStates from '@zooniverse/async-states'
+import React from 'react'
+import PropTypes from 'prop-types'
+
+import * as d3 from 'd3'
+import exampleData from './example-planet-hunters-data.json'
+
+import LightCurveViewer from './LightCurveViewer'
+import locationValidator from '../../helpers/locationValidator'
+
+class LightCurveViewerContainer extends React.Component {
+  constructor () {
+    super()
+    this.state = {}
+    
+    this.svgContainer = React.createRef();
+    
+    //TODO: turn into variables?
+    this.width = 500
+    this.height = 500
+    
+    this.d3svg = null;
+    this.d3dataLayer = null;
+  }
+
+  componentDidMount () {
+    //Prepare the main SVG
+    //--------------------------------
+    this.d3svg = d3.select(this.svgContainer.current).append('svg')
+      .attr('class', 'light-curve-viewer')
+      //.attr('viewBox', `0 0 ${this.width} ${this.height}`)  //TODO: check
+      //.attr('style', 'width: 100%; height: 100%')  //TODO: check
+      .attr('width', this.width)
+      .attr('height', this.height)
+    
+    //Note: an alternative way to later select this node is
+    //`d3.select(this.d3svg.node())`. Don't use `d3.select(this.d3svg)`.
```

Comment:
The other way is to save a ref, as in the prototype. The Smashing Mag article @srallen posted shows how to achieve this: https://www.smashingmagazine.com/2018/02/react-d3-ecosystem/#d3-js-within-react

Reply:
The current D3 system does use React's ref systems to specify the root node, as you had previously suggested.

However, unlike the prototype, I'm not using the `_setRef()` method outlined in the Smashing Magazine article, but instead I use `React.createRef()` as specified by the [React documentation](https://reactjs.org/docs/refs-and-the-dom.html) for `v16.5.2`, which looks like:
```
class MyComponent extends React.Component {
  constructor(props) {
    super(props);
    this.myRef = React.createRef();
  }
  render() {
    return <div ref={this.myRef} />;
  }
}
```

Aside: this specific comment is made in the context that, e.g. `d3.select(this.d3svg.node()).append('rect')` is an alternative to `this.d3svg.append('rect')`. It's just an observation on working with D3, (and the errors+fixes I encountered blundering through it,) and I can remove this comment if it's more confusing than clarifying.